external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -3204,6 +3215,65 @@ fn exp(context: &mut Context, pe: Box<P::Exp>) -> Box<E::Exp> {
     Box::new(sp(loc, e_))
 }
 
+fn exp_cast(context: &mut Context, in_parens: bool, plhs: Box<P::Exp>, pty: P::Type) -> E::Exp_ {
+    use E::Exp_ as EE;
+    use P::Exp_ as PE;
+    fn ambiguous_cast(e: &P::Exp) -> bool {
+        match &e.value {
+            PE::Value(_)
+            | PE::Move(_, _)
+            | PE::Copy(_, _)
+            | PE::Name(_, _)
+            | PE::Call(_, _, _, _)
+            | PE::Pack(_, _, _)
+            | PE::Vector(_, _, _)
+            | PE::Block(_)
```

Comment:
Maybe ambiguous?

Reply:
Which? I thought they all read okay in tests but happy to switch any you are worried about